source/game/kart/KartDynamics.cc

```diff
@@ -7,34 +7,59 @@ KartDynamics::KartDynamics() {
 }
 
 void KartDynamics::init() {
-    m_pos.setZero();
-    m_speed.setZero();
-    m_fullRot = EGG::Quatf(1.0f, EGG::Vector3f::zero);
+    m_speedNorm = 0.0f;
+    m_gravity = -1.0f;
     m_top = EGG::Vector3f::ey;
+    m_forceUpright = true;
+    m_noGravity = false;
 }
 
-const EGG::Vector3f &KartDynamics::getPos() const {
+void KartDynamics::resetInternalVelocity() {
+    m_velocity.setZero();
+}
+
+void KartDynamics::calc(f32 dt, f32 /*maxSpeed*/, bool /*air*/) {
+    if (!m_noGravity) {
+        m_totalForce.y += m_gravity;
+    }
+
+    m_acceleration0 = m_totalForce;
+    m_externalVel += m_acceleration0 * dt;
+    m_externalVel *= 0.998f;
+    m_velocity =
+            m_externalVel * dt + m_internalVelocity + m_movingObjectVelocity + m_movingRoadVelocity;
```

Comment:
Perhaps we could rename these `m_extVel`, `m_intVel`, `m_movingObjVel`, and `m_movingRdVel` to put this all on one line, but also to condense future use. I don't think these abbreviations would affect readability given minimal context.

Reply:
Aside from `m_movingRdVel` (which can be `m_movingRoadVel`) these are fine abbreviations.